collector/netframework_clr.go

```diff
@@ -0,0 +1,1218 @@
+			[]string{"process"},
+			nil,
+		),
+		ContextBoundClassesLoaded: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, remotingSubsystem, "context_bound_classes_loaded"),
+			"Displays the current number of context-bound classes that are loaded.",
+			[]string{"process"},
+			nil,
+		),
+		ContextBoundObjects: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, remotingSubsystem, "context_bound_objects_total"),
+			"Displays the total number of context-bound objects allocated.",
+			[]string{"process"},
+			nil,
+		),
+		ContextProxies: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, remotingSubsystem, "context_proxies_total"),
+			"Displays the total number of remoting proxy objects in this process since it started.",
+			[]string{"process"},
+			nil,
+		),
+		Contexts: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, remotingSubsystem, "contexts"),
+			"Displays the current number of remoting contexts in the application.",
+			[]string{"process"},
+			nil,
+		),
+		TotalRemoteCalls: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, remotingSubsystem, "remote_calls_total"),
+			"Displays the total number of remote procedure calls invoked since the application started.",
+			[]string{"process"},
+			nil,
+		),
+
+		// .NET CLR Security metrics
+		NumberLinkTimeChecks: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, securitySubsystem, "link_time_checks_total"),
+			"Displays the total number of link-time code access security checks since the application started.",
+			[]string{"process"},
+			nil,
+		),
+		TimeinRTchecks: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, securitySubsystem, "rt_checks_time_percent"),
+			"Displays the percentage of time spent performing runtime code access security checks in the last sample.",
+			[]string{"process"},
+			nil,
+		),
+		StackWalkDepth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, securitySubsystem, "stack_walk_depth"),
+			"Displays the depth of the stack during that last runtime code access security check.",
+			[]string{"process"},
+			nil,
+		),
+		TotalRuntimeChecks: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, securitySubsystem, "runtime_checks_total"),
+			"Displays the total number of runtime code access security checks performed since the application started.",
+			[]string{"process"},
+			nil,
+		),
+
+		// Process name whitelist and blacklist
+		processWhitelistPattern: regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *netclrWhitelist)),
+		processBlacklistPattern: regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *netclrBlacklist)),
+	}
+
+	netclrCollector.netclrCollectors = netclrCollector.getNETCLRCollectors()
+	if *netclrPrintCollectors {
+		fmt.Printf("Available .NET CLR sub-collectors:\n")
+		for name := range netclrCollector.netclrCollectors {
+			fmt.Printf(" - %s\n", name)
+		}
+		os.Exit(0)
+	}
```

Comment:
Perhaps move this to the top of the initialization?

Reply:
Resolved in latest push. I've moved it to the top of the `newNETCLRCollector()` function.